flow/Error.h

```diff
@@ -129,6 +129,33 @@ extern bool isAssertDisabled(int line);
 	} while (false)
 #define UNREACHABLE()                                                                                                  \
 	{ throw internal_error_impl("unreachable", __FILE__, __LINE__); }
+#define ASSERT_THROW(expression, exception)                                                                            \
+	do {                                                                                                               \
+		bool _____exception_happened = false;                                                                          \
+		try { expression; }                                                                                            \
+		catch(exception& ex) { _____exception_happened = true; }                                                       \
+		if (!_____exception_happened || isAssertDisabled(__LINE__)) {                                                  \
```

Comment:
This seems strange. Should it be `if (_____exception_happened && !isAssertDisabled(__LINE__)` then throw the error?

Basically I am confused what does `ASSERT_THROW` mean. Please add some comments on what these macros do.